src/Microsoft.NetCore.Analyzers/Core/Security/UseSecureSharedAccesSignature.cs

```diff
@@ -0,0 +1,199 @@
+                isEnabledByDefault: DiagnosticHelpers.EnabledByDefaultIfNotBuildingVSIX,
+                description: s_Description,
+                helpLinkUri: null,
+                customTags: WellKnownDiagnosticTags.Telemetry);
+        }
+
+        internal static ImmutableArray<(string, string)> NamespaceAndpolicyIdentifierNamePairs = ImmutableArray.Create(
+                                                                                                    ("Blob", "groupPolicyIdentifier"),
+                                                                                                    ("File", "groupPolicyIdentifier"),
+                                                                                                    ("Queue", "accessPolicyIdentifier"),
+                                                                                                    ("Table", "accessPolicyIdentifier"));
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(
+                                                                                        DoNotUseAccountSASRule,
+                                                                                        UseSharedAccessProtocolHttpsOnlyRule,
+                                                                                        UseContainerLevelAccessPolicyRule);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+
+            // Security analyzer - analyze and report diagnostics on generated code.
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.Analyze | GeneratedCodeAnalysisFlags.ReportDiagnostics);
+
+            context.RegisterCompilationStartAction(compilationStartAnalysisContext =>
+            {
+                compilationStartAnalysisContext.RegisterOperationBlockStartAction(operationBlockStartContext =>
+                {
+                    if (!(operationBlockStartContext.OwningSymbol is IMethodSymbol containingMethod))
+                    {
+                        return;
+                    }
+
+                    var interproceduralAnalysisConfig = InterproceduralAnalysisConfiguration.Create(
+                                                            operationBlockStartContext.Options,
+                                                            SupportedDiagnostics,
+                                                            defaultInterproceduralAnalysisKind: InterproceduralAnalysisKind.None,
+                                                            cancellationToken: operationBlockStartContext.CancellationToken,
+                                                            defaultMaxInterproceduralMethodCallChain: 1);
+                    var lazyValueContentResult = new Lazy<(DataFlowAnalysisResult<ValueContentBlockAnalysisResult, ValueContentAbstractValue>, PointsToAnalysisResult)>(
```

Comment:
>Lazy [](start = 53, length = 4)

Just FYI, DataFlowAnalysis.GetOrComputeResultCore() caches results so DFA doesn't have to be recomputed.

Naming tuple elements can make the code easier to read, so you don't have to use "Item1" and "Item2" below.  E.g.:
    Lazy<(DataFlowAnalysisResult<ValueContentBlockAnalysisResult, ValueContentAbstractValue> ValueContentResult, PointsToAnalysisResult PointsToResult)>